config/routes.rb

```diff
@@ -10,4 +10,7 @@
   end
   root :to => redirect('/users/sign_in')
   devise_for :users
+  patch 'users/:user_id/add_case/:id', to: 'users#add_case'
```

Comment:
Can you move these additional routes into the `authenticate :user do` block? Want to signify that they require being authenticated to do any of them.